packages/gatsby/src/schema/data-tree-utils.js

```diff
@@ -14,24 +23,36 @@ const flatten = require(`flat`)
  *
  * @param {*Nodes} args
  */
-export const extractFieldExamples = (nodes: any[]) => {
+const extractFieldExamples = (nodes: any[]) => {
   // $FlowFixMe
-  return _.mergeWith({}, ...nodes, (obj, next) => {
-    if (!_.isArray(obj || next)) return
-    let array = [].concat(obj, next).filter(v => v != null)
+  return _.mergeWith({}, ...nodes, (obj, next, key, o, s, stack) => {
+    if (obj === INVALID_VALUE) return obj
+
+    // TODO: if you want to support infering Union types this should be handled
+    // differently. Maybe merge all like types into examples for each type?
+    // e.g. union: [1, { foo: true }, ['brown']] -> Union Int|Object|List
+    if (!isSameType(obj, next)) return INVALID_VALUE
```

Comment:
we can always revisit, at the moment it's just not breaking on union code which should be ok for now. I'm not sure if stuff is still not building tho looking at the PR checks....stuff is running locally for me.